neqo-crypto/src/aead.rs

```diff
@@ -13,13 +13,75 @@ use std::{
 
 use crate::{
     constants::{Cipher, Version},
-    err::Res,
+    err::{sec::SEC_ERROR_BAD_DATA, Error, Res},
     experimental_api,
     p11::{PK11SymKey, SymKey},
     scoped_ptr,
     ssl::{PRUint16, PRUint64, PRUint8, SSLAeadContext},
 };
 
+/// Trait for AEAD (Authenticated Encryption with Associated Data) operations.
+///
+/// This trait provides a common interface for both real and null AEAD implementations,
+/// eliminating code duplication and allowing for consistent usage patterns.
+pub trait Aead {
+    /// Create a new AEAD instance.
+    ///
+    /// # Errors
+    ///
+    /// Returns `Error` when the underlying crypto operations fail.
+    fn new(version: Version, cipher: Cipher, secret: &SymKey, prefix: &str) -> Res<Self>
+    where
+        Self: Sized;
```

Comment:
Why is the explicit `Sized` needed?

Reply:
For this, it's because the function returns `Self`.